code/datums/components/food/edible.dm

```diff
@@ -212,10 +212,12 @@ Behavior that's still missing from this component that original food items had t
 
 	for(var/rid in reagents)
 		var/amount = reagents[rid]
-		if(length(tastes) && (rid == /datum/reagent/consumable/nutriment || rid == /datum/reagent/consumable/nutriment/vitamin))
-			owner.reagents.add_reagent(rid, amount, tastes.Copy(), added_purity = reagent_purity)
-		else
-			owner.reagents.add_reagent(rid, amount, added_purity = reagent_purity)
+		if(length(tastes) && ispath(rid, /datum/reagent/consumable/nutriment))
+			var/datum/reagent/consumable/nutriment/nid = rid
+			if(nid.carry_food_tastes)
```

Comment:
Should this be `iniital(nid.carry_food_tastes)`? It looks like `rid` is a typepath

Reply:
works as is, buuuuuuuut ...yeah probably